### PR TITLE
(WIP) Type Declaration for Header plugin

### PR DESCRIPTION
**Pull request by shvkhzod**
- Created tsconfig.ts , header.d.ts and index.ts
- Added types section in package.json

### CODE REVIEW

File: types/header.d.ts
Comment (ilyamore88) [inline types/header.d.ts:1]: You don't need to create .d.ts file by yourself. tsc will create it after compiling.

**Review by ilyamore88 — COMMENTED** [review]
Please, format your code using EsLint.

**Comment by ilyamore88** [comment]
Because of using Vite, your d.ts file might be missed after compiling. There is a plugin that can copy declaration files to the dist folder.

**Comment by neSpecc** [comment]
Please, update eslint config to use `codex/ts`

**Comment by neSpecc** [comment]
Resolved by #108

**Pull request closed without merging**